src/app/components/media/RestoreProjectMedia.js

```diff
@@ -0,0 +1,167 @@
+import { FlashMessageSetterContext } from '../FlashMessage';
+import UpdateProjectMediaMutation from '../../relay/mutations/UpdateProjectMediaMutation';
+import CheckArchivedFlags from '../../CheckArchivedFlags';
+
+function handleRestore({
+  team,
+  projectMedia,
+  context,
+  onSuccess,
+  onFailure,
+}) {
+  const newContext = context;
+  if (context.team && !context.team.public_team) {
+    newContext.team.public_team = Object.assign({}, context.team);
+  }
+
+  Relay.Store.commitUpdate(
+    new UpdateProjectMediaMutation({
+      id: projectMedia.id,
+      media: projectMedia,
+      archived: CheckArchivedFlags.NONE,
+      check_search_team: team.search,
+      check_search_project: null,
+      check_search_trash: team.check_search_trash,
+      check_search_spam: team.check_search_spam,
+      context: newContext,
+      srcProj: null,
+      dstProj: null,
+    }),
+    { onSuccess, onFailure },
+  );
+}
+
+function RestoreProjectMedia({
+  team,
+  projectMedia,
+  context,
+  className,
+}) {
+  const [isSaving, setIsSaving] = React.useState(false);
+  const setFlashMessage = React.useContext(FlashMessageSetterContext);
+
+  const handleSubmit = React.useCallback((project) => {
+    handleRestore({
+      team,
+      project,
+      projectMedia,
+      context,
+      onSuccess: () => {
+        setIsSaving(false);
+        const message = projectMedia.archived === CheckArchivedFlags.TRASHED ?
+          (
+            <FormattedMessage
+              id="mediaActionsBar.movedRestoreBack"
+              defaultMessage="Item restored from Trash"
+              description="Success message after restoring item from Trash."
+            />
+          ) :
+          (
+            <FormattedMessage
+              id="mediaActionsBar.movedSpamBack"
+              defaultMessage="Item marked as Not Spam"
+              description="Success message after marking item as Not Spam."
+            />
+          );
+        setFlashMessage(message, 'success');
+        window.location.assign(window.location.pathname);
+      },
+      onFailure: () => {
+        const message = projectMedia.archived === CheckArchivedFlags.TRASHED ?
+          (
+            <FormattedMessage
+              id="mediaActionsBar.failedmovedRestoreBack"
+              defaultMessage="Failed to restore the item. Please try again later."
+              description="Failure message after attempting to restore an item from Trash."
+            />
+          ) :
+          (
+            <FormattedMessage
+              id="mediaActionsBar.failedmovedSpamBack"
+              defaultMessage="Failed to mark the item as Not Spam. Please try again later."
+              description="Failure message after attempting to mark an item as Not Spam."
+            />
+          );
+        setFlashMessage(message, 'failure');
+        window.location.assign(window.location.pathname);
+        setFlashMessage('Failed to restore the item. Please try again later.', 'error');
```

Comment:
Please review this: use the translatable message with the correct message type (I think 'error')